BUILD

```diff
@@ -4,3 +4,28 @@ exports_files([
     "VERSION",
     ".clang-format",
 ])
+
+# These two definitions exist to help reduce Envoy upstream core code depending on extensions.
+# To avoid visibility problems, one can replace the package lists below with "//visibility:public"
+#
+# TODO(#9953) //test/config_test:__pkg__ should probably be split up and removed.
+# TODO(#9953) the config fuzz tests should be moved somewhere local and //test/config_test and //test/server removed.
+package_group(
+    name = "extension_config",
+    packages = [
+        "//source/exe",
+        "//source/extensions/...",
+        "//test/config_test",
+        "//test/extensions/...",
+        "//test/server",
+        "//test/server/config_validation",
+    ],
+)
+
+package_group(
+    name = "extension_library",
+    packages = [
+        "//source/extensions/...",
+        "//test/extensions/...",
+    ],
+)
```

Comment:
Can we add a TODO here to make this override more easier? I think potentially we can embed part of this in `source/extensions/extensions_build_config.bzl` (something like `additional_visibility`), that's where the overrides happen for downstream builds normally.

Reply:
Gah, that's a fair point.
I hate to make folks overwrite the build file, then move their workarounds when I do the TODO.  Let me see if I can get something working tonight.